### PR TITLE
Branch bug fix 1

### PR DESCRIPTION
**Pull request by Jacky142857**

### CODE REVIEW

File: src/main/java/seedu/address/model/client/Appointment.java [hunk 1]
```diff
@@ -48,7 +48,41 @@ public static boolean isValidAppointment(String time) {
...
+    /**
+     * Checks whether the input date is a valid date
+     */
+
+    private static boolean isValidDate(int year, int month, int day) {
+        if (month == 2 && isLeapYear(year)) {
+            return 1 <= day && day <= 29;
+        } else if (month == 2 && !isLeapYear(year)) {
+            return 1 <= day && day <= 28;
+        } else if ((month == 1) || (month == 3) || (month == 5) || (month == 7)
```
Comment (janald99) [inline src/main/java/seedu/address/model/client/Appointment.java:80]: Nice thorough checks, should be full proof now

**Review by janald99 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ public static Appointment parseAppointment(String appointment) throws ParseExcep
         String trimmedAppointment = appointment.trim();
         if (trimmedAppointment.equals("reset")) {
             return new Appointment("");
-        } else if (!Appointment.isValidAppointment(trimmedAppointment)) {
+        } else if (appointment.equals("") || !Appointment.isValidAppointment(trimmedAppointment)) {
```
Comment (WJunHong) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:82]: Good check

**Review by WJunHong — APPROVED** [review]
LGTM!

**Pull request merged**